lib/turn.rb

```diff
@@ -0,0 +1,68 @@
+class Turn
+    attr_reader :player1, :player2, :spoils_of_war
+    def initialize(player1, player2)
+        @player1 = player1
+        @player2 = player2
+        @spoils_of_war = []
+        @turn_type = nil
+    end
+
+    def type
+        if @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+            @turn_type = :basic
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0) && @player1.deck.cards.length < 3 || @player2.deck.cards.length < 3
```

Comment:
I like that you include another condition to check if each player even has enough cards to continue before determining if it is a `:mutually_assured_destruction` or `:war` situation. I know this was giving a lot of people fits at the end when running the game.